nodejs/awsx/ecsx/fargateTaskDefinition.ts

```diff
@@ -0,0 +1,394 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as aws from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+import {
+    defaultLogGroup,
+    DefaultLogGroupArgs,
+    LogGroupId,
+} from "../cloudwatch/logGroup";
+import * as role from "../role";
+import { DefaultRoleWithPolicyArgs } from "../role";
+import * as utils from "../utils";
+import { Container } from "./container";
+import { calculateFargateMemoryAndCPU } from "./fargateMemoryAndCpu";
+
+export interface FargateTaskDefinitionArgs
+    extends Omit<
+        aws.ecs.TaskDefinitionArgs,
+        | "containerDefinitions"
+        | "cpu"
+        | "executionRoleArn"
+        | "family"
+        | "memory"
+        | "taskRoleArn"
+    > {
+    /**
+     * Single container to make a TaskDefinition from.  Useful for simple cases where there aren't
+     * multiple containers, especially when creating a TaskDefinition to call [run] on.
+     *
+     * Either [container] or [containers] must be provided.
+     */
+    container?: Container;
```

Comment:
What do you mean by a prompt type? 

This is just a plain input. The fields inside this type are inputs.

It's concrete rather than `Input<Container>` for simplicity and carrying throught from the old implementation. Though, looking again, we could quite easily make these two an optional input type.